### ISSUE TITLE
World age assertion in codegen

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#29267]
After pulling master today, my current project is giving the following assertion error:
```
Assertion `li->min_world <= world && li->max_world >= world' failed.
```
Cursory debugging suggests the method has been deleted in the current world. Investigating further.
```
(rr) p li->min_world
$9 = 25116
(rr) p world
$11 = 28906
(rr) p li->max_world
$14 = 28839
```

**Comment by timholy** [comment]
Were you using Revise? If so, and if you hadn't updated packages in a while, it could have been https://github.com/timholy/Revise.jl/issues/191.

**Comment by Keno** [comment]
Revise was loaded, but the method in question was not deleted by Revise. Rather it got invalidated when another package was loaded.

**Comment by vtjnash** [comment]
This is pretty old now, so I'm going to presume it fixed by #31191 and see if any new issues get opened as they pop up.
